### PR TITLE
fix code coverage calculation

### PR DESCRIPTION
**Pull request by sbcd90**
### Description
improves code coverage to 80% by including coverage from integration tests.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by praveensameneni — DISMISSED** [review]

File: build-tools/opensearchplugin-coverage.gradle [hunk 1]
```diff
@@ -3,24 +3,26 @@
...
+
 /**
- * OpenSearch Plugin build tools don't work with the Gradle Jacoco Plugin to report coverage out of the box.
+ * ES Plugin build tools don't work with the Gradle Jacoco Plugin to report coverage out of the box.
  * https://github.com/elastic/elasticsearch/issues/28867.
  *
- * This code sets up coverage reporting manually for OpenSearch plugin tests. This is complicated because:
- *  1. The OpenSearch integTest Task doesn't implement Gradle's JavaForkOptions so we have to manually start the jacoco agent with the test JVM
+ * This code sets up coverage reporting manually for ES plugin tests. This is complicated because:
+ *  1. The ES integTest Task doesn't implement Gradle's JavaForkOptions so we have to manually start the jacoco agent with the test JVM
```
Comment (engechas) [inline build-tools/opensearchplugin-coverage.gradle:13]: We should revert these naming changes

File: build-tools/opensearchplugin-coverage.gradle [hunk 2]
```diff
@@ -3,24 +3,26 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+apply plugin: 'jacoco'
+
 /**
- * OpenSearch Plugin build tools don't work with the Gradle Jacoco Plugin to report coverage out of the box.
+ * ES Plugin build tools don't work with the Gradle Jacoco Plugin to report coverage out of the box.
```
Comment (eirsep) [inline build-tools/opensearchplugin-coverage.gradle:9]: why did we change this to ES from opensearch?

File: build-tools/opensearchplugin-coverage.gradle [hunk 3]
```diff
@@ -3,24 +3,26 @@
...
  * https://github.com/elastic/elasticsearch/issues/28867.
  *
- * This code sets up coverage reporting manually for OpenSearch plugin tests. This is complicated because:
- *  1. The OpenSearch integTest Task doesn't implement Gradle's JavaForkOptions so we have to manually start the jacoco agent with the test JVM
+ * This code sets up coverage reporting manually for ES plugin tests. This is complicated because:
```
Comment (eirsep) [inline build-tools/opensearchplugin-coverage.gradle:12]: why did we change this to ES from opensearch?

File: src/test/java/org/opensearch/securityanalytics/SecurityAnalyticsRestTestCase.java [hunk 4]
```diff
@@ -1786,4 +1794,43 @@ public String getMatchAllSearchRequestString(int num) {
...
+
+        void dump(boolean reset);
+
+        void reset();
+    }
+
+
+    @AfterClass
+    public static void dumpCoverage() throws IOException, MalformedObjectNameException {
+        // jacoco.dir is set in esplugin-coverage.gradle, if it doesn't exist we don't
```
Comment (engechas) [inline src/test/java/org/opensearch/securityanalytics/SecurityAnalyticsRestTestCase.java:1817]: esplugin-coverage.gradle -> opensearchplugin-coverage.gradle

**Review by engechas — APPROVED** [review]

**Review by praveensameneni — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Pull request merged**